api-tests/src/steps/cri-steps.ts

```diff
@@ -72,6 +76,54 @@ When(
   },
 );
 
+// This step sends a request to a CRI stub and then processes that response in core back. It also validates that
+// the initial request to the CRI stub contained the specified attributes. These attributes are encrypted so we
+// have to wait for the CRI stub to decrypt them and send them back to the test code rather than just validating
+// CRI stub request directly.
+When(
+  /^I submit (expired )?'([\w-]+)' details with attributes to the (async )?CRI stub$/,
+  async function (
+    this: World,
+    expired: "expired " | undefined,
+    scenario: string,
+    async: "async " | undefined,
+    dataTable: DataTable | undefined,
+  ): Promise<void> {
+    if (!this.lastJourneyEngineResponse) {
+      throw new Error("No last journey engine response found.");
+    }
+
+    if (!isCriResponse(this.lastJourneyEngineResponse)) {
+      throw new Error("Last journey engine response was not a CRI response");
+    }
+
+    const jarPayload = await submitAndProcessCriAction(
+      this,
+      await generateCriStubBody(
+        this.lastJourneyEngineResponse.cri.id,
+        scenario,
+        this.lastJourneyEngineResponse.cri.redirectUrl,
+        expired ? EXPIRED_NBF : undefined,
+        !!async,
+      ),
+    );
+
+    if (jarPayload && dataTable?.rows) {
```

Comment:
When would there be no `jarPayload` - should that be an error?

Reply:
Looks like it should be, yes